playbooks/common/openshift-cluster/upgrades/v3_5/validator.yml

```diff
@@ -62,6 +62,6 @@
         $ oc get petsets --all-namespaces -o yaml | oc delete -f - --cascale=false
     when:
     # Search did not fail, valid resource type found
-    - l_do_petsets_exist.results.returncode == "0"
+    - l_do_petsets_exist.results.returncode == 0
```

Comment:
This change seems sane to me, if there is a chance of the value being a string rather than an int (set_fact is generally a culprit for those cases), then you can always use the `int` filter to do an explicit conversion.